src/main/java/com/direwolf20/buildinggadgets/client/gui/base/EntryList.java

```diff
@@ -0,0 +1,197 @@
+
+        renderTransition(tessellator, bufferbuilder, y0, y0 + 4);
+        renderTransition(tessellator, bufferbuilder, y1 - 4, y1);
+        GlStateManager.shadeModel(GL_FLAT);
+
+        int j1 = getMaxScroll();
+        if (j1 > 0) {
+            int k1 = (int) ((float) ((y1 - y0) * (y1 - y0)) / (float) getMaxPosition());
+            k1 = MathHelper.clamp(k1, 32, y1 - y0 - 8);
+            int l1 = (int) getScrollAmount() * (y1 - y0 - k1) / j1 + y0;
+            if (l1 < y0) {
+                l1 = y0;
+            }
+            int x1 = getScrollbarPosition();
+            int x2 = x1 + 6;
+
+            GlStateManager.disableTexture();
+            bufferbuilder.begin(GL_QUADS, DefaultVertexFormats.POSITION_COLOR);
+            bufferbuilder.pos(x1, y1, 0.0D).color(0, 0, 0, 255).endVertex();
+            bufferbuilder.pos(x2, y1, 0.0D).color(0, 0, 0, 255).endVertex();
+            bufferbuilder.pos(x2, y0, 0.0D).color(0, 0, 0, 255).endVertex();
+            bufferbuilder.pos(x1, y0, 0.0D).color(0, 0, 0, 255).endVertex();
+
+            bufferbuilder.pos(x1, (l1 + k1), 0.0D).color(128, 128, 128, 255).endVertex();
+            bufferbuilder.pos(x2, (l1 + k1), 0.0D).color(128, 128, 128, 255).endVertex();
+            bufferbuilder.pos(x2, l1, 0.0D).color(128, 128, 128, 255).endVertex();
+            bufferbuilder.pos(x1, l1, 0.0D).color(128, 128, 128, 255).endVertex();
+
+            bufferbuilder.pos(x1, (l1 + k1 - 1), 0.0D).color(192, 192, 192, 255).endVertex();
+            bufferbuilder.pos((x2 - 1), (l1 + k1 - 1), 0.0D).color(192, 192, 192, 255).endVertex();
+            bufferbuilder.pos((x2 - 1), l1, 0.0D).color(192, 192, 192, 255).endVertex();
+            bufferbuilder.pos(x1, l1, 0.0D).color(192, 192, 192, 255).endVertex();
+            tessellator.draw();
+        }
+
+        renderDecorations(mouseX, mouseX);
+        GlStateManager.enableTexture();
+        GlStateManager.shadeModel(GL_FLAT);
+        GlStateManager.enableAlphaTest();
+        GlStateManager.disableBlend();
+    }
+
+    protected void renderTransition(Tessellator tessellator, BufferBuilder buffer, int transitionTop, int transitionBottom) {
+        GlStateManager.enableBlend();
+        GlStateManager.blendFuncSeparate(SourceFactor.SRC_ALPHA, DestFactor.ONE_MINUS_SRC_ALPHA, SourceFactor.ZERO, DestFactor.ONE);
+        GlStateManager.disableAlphaTest();
+        GlStateManager.shadeModel(GL_SMOOTH);
+        buffer.begin(GL_QUADS, DefaultVertexFormats.POSITION_COLOR);
+        buffer.pos(x0, transitionTop, 0.0D).color(0, 0, 0, 255).endVertex();
+        buffer.pos(x1, transitionTop, 0.0D).color(0, 0, 0, 255).endVertex();
+        buffer.pos(x1, transitionBottom, 0.0D).color(0, 0, 0, 0).endVertex();
+        buffer.pos(x0, transitionBottom, 0.0D).color(0, 0, 0, 0).endVertex();
+        tessellator.draw();
+    }
+
+    protected void renderContentBackground(Tessellator tessellator, BufferBuilder bufferbuilder) {
```

Comment:
Why do we have a method for a single line when the method is protected stopping us from Overriding it if we so wanted to.